src/thunderbots/software/network_input/networking/network_client.cpp

```diff
@@ -0,0 +1,131 @@
+#include "network_input/networking/network_client.h"
+
+#include <boost/bind.hpp>
+
+#include "util/constants.h"
+#include "util/logger/init.h"
+#include "util/ros_messages.h"
+
+NetworkClient::NetworkClient(ros::NodeHandle& node_handle) : backend(), io_service()
+{
+    // Set up publishers
+    world_publisher = node_handle.advertise<thunderbots_msgs::World>(
+        Util::Constants::NETWORK_INPUT_WORLD_TOPIC, 1);
+    gamecontroller_publisher = node_handle.advertise<thunderbots_msgs::RefboxData>(
+        Util::Constants::NETWORK_INPUT_GAMECONTROLLER_TOPIC, 1);
+
+    // Set up our connection over udp to receive vision packets
+    // NOTE: We do this before initializing the ROS node so that if it
+    // fails because there is another instance of this node running
+    // and connected to the port we want, we don't kill that other node.
```

Comment:
Do we? Does this class own the ROS stuff?

Reply:
Ah this isn't applicable anymore. 

ROS nodes will shut down if a new node is registered with the same name. We need to initialize the ROS node to get the NodeHandle before creating the NetworkClient anyway, so we can't attempt to bind to the network port before registering the node. This isn't that important anyway.

Comments removed.